pkg/archive/tar.go

```diff
@@ -157,6 +157,31 @@ func Apply(ctx context.Context, root string, r io.Reader, opts ...ApplyOpt) (int
 	return options.applyFunc(ctx, root, r, options)
 }
 
+// cachedRootPath will memoize root paths, avoiding redundant checks.
+type cachedRootPath struct {
+	root  string
+	cache map[string]string
+}
+
+func newCachedRootPath(root string) *cachedRootPath {
+	return &cachedRootPath{
+		root:  root,
+		cache: make(map[string]string),
+	}
+}
+
+func (c cachedRootPath) Get(path string) (string, error) {
```

Comment:
`(c cachedRootPath)` -> `(c *cachedRootPath)`; `Get` -> `get`

Reply:
oh, yes, updated !

Reply:
@azr can you also update `func (c cachedRootPath)` to `func (c *cachedRootPath)`? Otherwise I think every call will incur a copy of `cachedRootPath` which is unnecessary. thanks

Reply:
ah extremely good point haha, will test this a bit as well.

Reply:
okay, looking good, saw stuff being cached and pull worked.